.github/workflows/tests-contributors.yml

```diff
@@ -37,30 +37,73 @@ jobs:
+          else
+            echo "Branch $TARGET_BRANCH should not run tests, aborting."
+            exit 0
+          fi
+
+          # ...here "release" in first array element is replaced with empty string to make the displayed name always the same.
+          JSON_REPLACED=$(echo "$JSON_MERGED" | jq -c '.[0].release |= ""')
+          # Turn "release" values into an array to be used as the primary key for matrix.
+          # Do not just use the whole JSON directly as include, it generates job names with all variables.
+          OUT_JSON_PRIMARY=$(echo "$JSON_REPLACED" | jq -c '[.[].release]')
+          # To use the whole JSON definition, pass the array to include to get what include: does.
+          OUT_JSON_SECONDARY="$JSON_REPLACED"
+
+          # Pretty-print the results for debug and set the outputs
+          echo "Generated JSONs are:"
+          echo "$OUT_JSON_PRIMARY" | jq '.'
+          echo "$OUT_JSON_SECONDARY" | jq '.'
+          echo "::set-output name=matrix-primary::$OUT_JSON_PRIMARY"
+          echo "::set-output name=matrix-secondary::$OUT_JSON_SECONDARY"
+
     outputs:
       allowed_user: ${{ steps.check_user_perm.outputs.allowed_user }}
       base_ref: ${{ fromJson(steps.pr_api.outputs.data).base.ref }}
       sha: ${{ fromJson(steps.pr_api.outputs.data).head.sha }}
+      matrix-primary: ${{ steps.generate_matrix.outputs.matrix-primary }}
+      matrix-secondary: ${{ steps.generate_matrix.outputs.matrix-secondary }}
 
-  unit-tests-master:
+  unit-tests-fedora:
     needs: pr-info
-    if: needs.pr-info.outputs.base_ref == 'master' && needs.pr-info.outputs.allowed_user == 'true'
-    runs-on: ubuntu-20.04
+    if: needs.pr-info.outputs.matrix-primary != '' && needs.pr-info.outputs.allowed_user == 'true'
+    runs-on: ubuntu-latest
     timeout-minutes: 30
     strategy:
       fail-fast: false
       matrix:
-        # empty = release that corresponds to current branch name
-        release: ['', 'eln']
-        include:
-          - release: eln
-            name-postfix: ' (eln)'
-            build-args: '--build-arg=image=quay.io/fedoraci/fedora:eln-x86_64'
+        # Empty release corresponds to current branch name. Name of the job with no suffix is set
+        # as the required check, so that it must always exist. See also matrix generation earlier.
+        # Using release: separately is needed for pretty job names that fit the scheme.
+        # The additional values for each release value come via the include: line.
+        release: ${{ fromJson( needs.pr-info.outputs.matrix-primary ) }}
+        include: ${{ fromJson( needs.pr-info.outputs.matrix-secondary ) }}
     env:
-      CI_TAG: '${{ matrix.release }}'
+      CI_TAG: '${{ matrix.ci-tag }}'
       CONTAINER_BUILD_ARGS: '${{ matrix.build-args }}'
-      STATUS_NAME: 'unit-tests${{ matrix.name-postfix }}'
-      TARGET_BRANCH_NAME: origin/master
+      STATUS_NAME: 'unit-tests (${{ matrix.release }})'
```

Comment:
This will not work as expected. This will name the status on PR for main run as:

```
unit-tests ()
```

which won't match required test `unit-tests`.

Reply:
Following the discussion: Re-introduce `name-postfix`. It will contain the name in brackets and will be the thing to clear in JSON instead of `release`. Then, the primary JSON can be dropped. The reason why nice job names do not matter is that they do *not* appear in the PR check list, we add these manually via direct API call and control the name there (`context`).